### ISSUE TITLE
Serializer should throw NotNormalizableValueException instead of InvalidArgumentException when an invalid enum is deserialized

### ISSUE THREAD

**Issue by ludekbenedik** [symfony/symfony#49711]
### Description

Hi,
when deserialize data to an entity with an enum attribute and data contains an invalid enum 's value, the serializer throws `InvalidArgumentException` which is a logic exception.

1. This is not a program logic error so `InvalidArgumentException` is not correct.
2. `InvalidArgumentException` is very a common exception so catching it is very difficult. A programmer must test the exception's message if an other logic exception have to be threw and logged.

For example, when deserialize a wrong date format then `NotNormalizableValueException` is thrown. For a programmer these situations are equal.

### Example

```php
enum MyEnum: int
{
    case A = 0;
    case B = 1;
}

class MyEntity
{
    public MyEnum $myEnum;
}

$entity = $serializer->deserialize('{"myEnum":2}', MyEntity::class, 'json')
```

### Before:

Throws `InvalidArgumentException`

### After:

Throws `NotNormalizableValueException`

**Comment by Gwemox** [comment]
Looks like it was changed by https://github.com/symfony/symfony/pull/47128

I also have a related bug https://github.com/symfony/symfony/issues/47797

**Comment by secit-pl** [comment]
In my opinion we should throw NotNormalizableValueException instead of InvalidArgumentException. At this moment it's not possible to use backed enums in real world app while deserializing invalid data from user (user provide wrong value to backed enum setter). InvalidArgumentException does not have methods to return the detailed error message to user that the allowed values are `x`, `y`, `z` but the provided value is `k`, we can only return "Hey, something goes wrong, guess what!", without any details.

**Comment by fabpot** [comment]
Should be fixed by #51475
